File: module-1/Week3_Review/exercise-final/dotnet/PetElevator/CRM/Customer.cs
using System.Collections.Generic;
using PetElevator.Shared;

namespace PetElevator.CRM
{
    public class Customer : Person, IBillable
    {
        public string PhoneNumber { get; set; }
        public List<Pet> Pets { get; set; } = new List<Pet>();

        public Customer(string firstName, string lastName) : this(firstName, lastName, "")
        {
        }
        public Customer(string firstName, string lastName, string phoneNumber) : base(firstName, lastName)
        {
            PhoneNumber = phoneNumber;
        }

        public double GetBalanceDue(Dictionary<string, double> servicesRendered)
        {
            double owes = 0.0;
            foreach (KeyValuePair<string, double> sr in servicesRendered)
            {
                owes += sr.Value;
            }
            return owes;
        }
    }
}

File: module-1/Week3_Review/exercise-final/java/src/main/java/com/techelevator/crm/Customer.java
package com.techelevator.crm;

import com.techelevator.Billable;
import com.techelevator.Person;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class Customer extends Person implements Billable {

    private String phoneNumber;
    private List<Pet> pets = new ArrayList<Pet>();

    public Customer(String firstName, String lastName) {
        this(firstName,lastName,"");
    }

    public Customer(String firstName, String lastName, String phoneNumber) {
        super(firstName, lastName);
        this.phoneNumber = phoneNumber;
    }

    public double getBalanceDue(Map<String, Double> servicesRendered) {
        double owes = 0.0;
        for(Map.Entry<String,Double> sr : servicesRendered.entrySet()) {
            owes += sr.getValue();
        }

        return owes;
    }
}
